scripts/test_watch_nu_releases.py

```
@@ -0,0 +1,276 @@
+                self.assertEqual(self.mod.load_current_state(state_file).get("latest_version"), "0.115.0")
+
+                # 3. Subsequent run without new version returns 0 without re-writing
+                code = self.mod.process_release(release, write=True, dry_run=False, force=False, state_file=state_file, releases_md=md_file)
+                self.assertEqual(code, 0)
+
+                # 4. Force run succeeds
+                code = self.mod.process_release(release, write=True, dry_run=False, force=True, state_file=state_file, releases_md=md_file)
+                self.assertEqual(code, 0)
+
+    def test_process_release_invalid_tag_raises(self):
+        with self.assertRaises(ValueError):
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)